### PR TITLE
cudnn.fastest in functional

### PR DESCRIPTION
**Pull request by szagoruyko**
Makes functional interface use cudnn.getSharedWorkspace if cudnn.fastest set

### CODE REVIEW

**Comment by borisfom** [comment]
@szagoruyko : do we even need functional.lua? I was planning to retire it in cudnn convolutions refactoring as it's becoming non-maintainable, check out our fork at https://github.com/NVIDIA/torch-cudnn.

**Comment by szagoruyko** [comment]
@borisfom I use it often.

File: functional.lua [hunk 1]
```diff
@@ -103,23 +103,39 @@ cudnn.functional.Convolution2D_updateOutput = function(handle, input, weight, ou
...
+   local bufSize = torch.LongTensor(1)
+   errcheck('cudnnGetConvolutionForwardWorkspaceSize',
+              handle,
+              iDesc[0], weightDesc[0],
+              convDesc[0], oDesc[0],
+              algType[0], bufSize:data())
+   local maxBufSize = bufSize[1]
+
+   local extraBuffer = workspace or cudnn.getSharedWorkspace()
+   local extraBufferSizeInBytes = extraBuffer:nElement() * 4 -- extraBuffer is always float
```
Comment (ngimel) [inline functional.lua]: extraBuffer is float, but I think still safer to use :elementSize() here.

**Comment by borisfom** [comment]
@szagoruyko: thanks for the input! We may need to figure out then how it should be refactored to use common cudnnFind routines, as per my comment in https://github.com/soumith/cudnn.torch/pull/250

**Pull request merged**